app/src/main/java/com/amaze/filemanager/activities/MainActivity.java

```diff
@@ -1189,6 +1193,8 @@ protected void onDestroy() {
         utilsHandler.close();
         cloudHandler.close();
 
+        if (mCloudCursorData != null) mCloudCursorData.close();
```

Comment:
> [...]
> The Loader will release the data once it knows the application
> is no longer using it.  For example, if the data is
> a {@link android.database.Cursor} from a {@link android.content.CursorLoader},
> you should not call close() on it yourself.  If the Cursor is being placed in a
> {@link android.widget.CursorAdapter}, you should use the
> {@link android.widget.CursorAdapter#swapCursor(android.database.Cursor)}
> method so that the old Cursor is not closed.
> [...]

-- [onLoadFinished() documentation](https://developer.android.com/reference/android/app/LoaderManager.LoaderCallbacks.html#onLoadFinished(android.content.Loader%3CD%3E,%20D))

Reply:
We don't need the old cursor, this is just a hack to know if the request is a duplicate one. I couldn't find a proper solution to this for the time being.
Can safely close the cursor, as we're not actively using it in any adapter.

Reply:
Closing this cursor is undefined behaviour, you can't.

Reply:
What? It's the original cursor which extends this undefined behaviour and is automatically handled by LoaderManager. This is a different cursor, a copy of the original one. It's my responsibility to empty this.

Reply:
@1909 you are getting a strong reference, not a copy, it is literally the same cursor.